TWTValidation/JSON Validator/Processors/TWTJSONObjectValidatorGenerator.m

```diff
@@ -0,0 +1,410 @@
+
+- (void)processTopLevelNode:(TWTJSONSchemaTopLevelASTNode *)topLevelNode
+{
+    [topLevelNode.schema acceptProcessor:self];
+}
+
+
+- (void)processGenericNode:(TWTJSONSchemaGenericASTNode *)genericNode
+{
+    TWTValidator *commonValidator = [self commonValidatorFromNode:genericNode];
+    TWTValidator *typeValidator = nil;
+
+    if ([genericNode.validTypes containsObject:TWTJSONSchemaTypeKeywordBoolean]) {
+        typeValidator = [[TWTBlockValidator alloc] initWithBlock:^BOOL(id value, NSError *__autoreleasing *outError) {
+            return [value isKindOfClass:[NSNumber class]] && strcmp([(NSValue *)value objCType], @encode(BOOL)) == 0;
+        }];
+    } else if ([genericNode.validTypes containsObject:TWTJSONSchemaTypeKeywordNull]) {
+        typeValidator = [TWTValueValidator valueValidatorWithClass:[NSNull class] allowsNil:NO allowsNull:YES];
+    }
+
+    [self pushJSONObjectValidatorWithCommonValidator:commonValidator typeValidator:typeValidator node:genericNode];
+}
+
+
+- (void)processArrayNode:(TWTJSONSchemaArrayASTNode *)arrayNode
+{
+    TWTValidator *commonValidator = [self commonValidatorFromNode:arrayNode];
+
+    TWTValidator *itemValidator = nil;
+    NSArray *indexedItemValidators = nil;
+    if (arrayNode.itemSchema) {
+        itemValidator = [self validatorFromNode:arrayNode.itemSchema];
+    } else {
+        indexedItemValidators = [self validatorsFromNodeArray:arrayNode.indexedItemSchemas];
+    }
+
+    TWTValidator *additionalItemsValidator = [self validatorFromNode:arrayNode.additionalItemsNode];
+    TWTJSONSchemaArrayValidator *typeValidator = [[TWTJSONSchemaArrayValidator alloc] initWithMaximumItemCount:arrayNode.maximumItemCount
+                                                                                              minimumItemCount:arrayNode.minimumItemCount
+                                                                                           requiresUniqueItems:arrayNode.requiresUniqueItems
+                                                                                                 itemValidator:itemValidator
+                                                                                         indexedItemValidators:indexedItemValidators
+                                                                                      additionalItemsValidator:additionalItemsValidator];
+
+    [self pushJSONObjectValidatorWithCommonValidator:commonValidator typeValidator:typeValidator node:arrayNode];
+}
+
+
+- (void)processNumberNode:(TWTJSONSchemaNumberASTNode *)numberNode
+{
+    TWTValidator *commonValidator = [self commonValidatorFromNode:numberNode];
+
+    [self pushNewObject:[[NSMutableArray alloc] init]];
+
+    [self addSubvalidator:[[TWTBlockValidator alloc] initWithBlock:^BOOL(id value, NSError *__autoreleasing *outError) {
+        return [value isKindOfClass:[NSNumber class]] && strcmp([(NSValue *)value objCType], @encode(BOOL)) != 0;
+    }]];
```

Comment:
I’ve changed my mind. Just leave a comment saying what’s going on and cast to NSNumber instead.